packages/router/src/__tests__/util.test.ts

```diff
@@ -38,6 +38,72 @@ describe('matchPath', () => {
       params: { id: 1337 },
     })
   })
+
+  it('transforms a param for Boolean', () => {
+    expect(matchPath('/signedUp/{status:Boolean}', '/signedUp/true')).toEqual({
+      match: true,
+      params: {
+        status: true,
+      },
+    })
+
+    expect(matchPath('/signedUp/{status:Boolean}', '/signedUp/false')).toEqual({
+      match: true,
+      params: {
+        status: false,
+      },
+    })
+
+    expect(
+      matchPath('/signedUp/{status:Boolean}', '/signedUp/somethingElse')
+    ).toEqual({
+      match: false,
+    })
+  })
+
+  it('transforms a param for floats', () => {
+    expect(
+      matchPath('/version/{floatyMcFloat:Float}', '/version/1.58')
+    ).toEqual({
+      match: true,
+      params: {
+        floatyMcFloat: 1.58,
```

Comment:
These params are for you @cannikin to enjoy

Reply:
haha Approved!